bindings/wallet-cordova/src/ios/WalletPlugin.m

```diff
@@ -237,13 +246,25 @@ - (void)WALLET_SET_STATE:(CDVInvokedUrlCommand*)command
 
     NSString* wallet_ptr_raw = [command.arguments objectAtIndex:0];
     NSString* value_raw = [command.arguments objectAtIndex:1];
-    NSString* counter_raw = [command.arguments objectAtIndex:2];
+    NSArray* counters_raw = [command.arguments objectAtIndex:2];
```

Comment:
I don't know if this works tbh, as there is no other place in the code where we take a json array

Reply:
Well, this _should_ work so no objections here. If this part if covered by tests, I can check if it works later today.

Reply:
well, it should be covered, although if it fails the problem can be in either this function or in the function that gives you the spending counters, as the test sets the spending counters, creates a tx, and asserts over the new counters.